Pylint fixes and deprecation of rare used methods in Connection (#9419)
mistercrunch/Airflow: UPDATING.md

-->

### Deprecated method in Connection

The Connection class has new deprecated methods.

> - Connection.parse_from_uri
> - Connection.log_info
> - Connection.debug_info

Reviewer: ```suggestion
- `Connection.parse_from_uri`
- `Connection.log_info`
- `Connection.debug_info`
```

> You used to create a connection object in two ways

Reviewer: ```suggestion
Previously, users could create a connection object in two ways
```

```
conn_1 = Connection(conn_id="conn_a", uri="mysql://AAA/")
# or
conn_2 = Connection(conn_id="conn_a")
conn_2.parse_uri(uri="mysql://AAA/")
```
Now the second way is not supported.

> Instead of the log_info and debug_info method. You can read each field individually or use the
> default representation (`__repr__`).

Reviewer: ```suggestion
`Connection.log_info` and `Connection.debug_info` method have been deprecated. Read each Connection field individually or use the
default representation (`__repr__`).
```
